packages/shared/src/hooks/logging-hook.ts

```diff
@@ -0,0 +1,70 @@
+import type { OpenFeatureError } from '../errors';
+import type { BaseHook } from './hook';
+import type { BeforeHookContext, HookContext, HookHints } from './hooks';
+import type { FlagValue, EvaluationDetails } from '../evaluation';
+
+import { DefaultLogger, SafeLogger } from '../logger';
+
+type LoggerPayload = Record<string, unknown>;
+
+const DOMAIN_KEY = 'domain';
+const PROVIDER_NAME_KEY = 'provider_name';
+const FLAG_KEY_KEY = 'flag_key';
+const DEFAULT_VALUE_KEY = 'default_value';
+const EVALUATION_CONTEXT_KEY = 'evaluation_context';
+const ERROR_CODE_KEY = 'error_code';
+const ERROR_MESSAGE_KEY = 'error_message';
+const REASON_KEY = 'reason';
+const VARIANT_KEY = 'variant';
+const VALUE_KEY = 'value';
+
+export class LoggingHook implements BaseHook {
+  readonly includeEvaluationContext: boolean = false;
+  readonly logger = new SafeLogger(new DefaultLogger(true, true));
```

Comment:
Please allow a custom logger to be passed in the constructor. If one isn't provided, we can fall back on what you have here.

Reply:
@beeme1mr This has been addressed with latest commit.